tests/src/core/testqgsrasterlayer.cpp

```diff
@@ -997,5 +1004,19 @@ void TestQgsRasterLayer::sample()
   QVERIFY( !ok );
 }
 
+void TestQgsRasterLayer::testTemporalProperties()
+{
+  QCOMPARE( mpRasterLayer->temporalProperties(), temporalProperties );
+  QVERIFY( !mpRasterLayer->temporalProperties()->isActive() );
+
+  QgsDateTimeRange dateTimeRange = QgsDateTimeRange( QDateTime( QDate( 2020, 1, 1 ) ),
+                                   QDateTime( QDate( 2020, 12, 31 ) ) );
+
+  temporalProperties->setFixedTemporalRange( dateTimeRange );
+  QCOMPARE( mpRasterLayer->temporalProperties()->fixedTemporalRange(), dateTimeRange );
+
```

Comment:
Also add a test of the raster temporal properties writeXml and readXml methods (when you've implemented these)